replacer.go

```diff
@@ -112,7 +112,7 @@ func (r *Replacer) ReplaceFunc(input string, f ReplacementFunc) (string, error)
 func (r *Replacer) replace(input, empty string,
 	treatUnknownAsEmpty, errOnEmpty, errOnUnknown bool,
 	f ReplacementFunc) (string, error) {
-	if !strings.Contains(input, string(phOpen)) {
+	if !strings.Contains(input, string(phOpen)) && !strings.Contains(input, string(phClose)) {
```

Comment:
Interesting, was this needed for a test case to pass? I figured if there is no opening brace, there is definitely no placeholder -- we don't even have to check for a closing one.

Reply:
This was needed for cases like this:

```plain
{
	input:  `\}`,
	expect: `}`,
},
```

Without it, you don't get the chance to remove the escape character from a closing braces. It's a valid question as to whether this is a case we want to handle.

Reply:
That's definitely a good edge case... but one I'd be surprised if we ever hit in real life (at this point). Let's remove the extra check and gain back a couple nanoseconds. I think it's OK if input `\}` outputs `\}` unless we have users complaining otherwise...

Reply:
Will push a version with the extra check removed and tests modified accordingly. Benchmarks are in the same ballpark but I'll re-post for confirmation.